expo/expo
# Fix registering wrapper component provider

# Why

Fixes https://github.com/expo/expo/issues/6473

# How

Instead of blocking future attempts to register wrapper component provider and showing warning message, we just patch next provider component to make it always be wrapped inside our DevAppContainer component.

# Test Plan

1. Clone this test repo https://github.com/serhiipalash/test-expo-react-native-modals
2. Start the project in iOS Simulator
3. Verify that modals from `react-native-modals` library don't work
4. Verify that "Refreshed" popup raises when you change code and app is updated with RN Fast Refresh

5. Apply this pr patch
6. Verify that modals from `react-native-modals` library work as expected
7. Verify that "Refreshed" popup raises when you change code and app is updated with RN Fast Refresh

Follow this video https://drive.google.com/file/d/1-KU56oP4xAkAeNA_NV__W6vg6ZLMvcRC/view and this comment https://github.com/expo/expo/issues/6473#issuecomment-565761931
